viadot/sources/spectrum.py

```diff
@@ -0,0 +1,166 @@
+from typing import List, Dict, Literal
+import boto3
+
+import pandas as pd
+import awswrangler as wr
+
+
+class Spectrum:
+    """
+    A class for pulling data from and uploading to Spectrum.
+
+    Args:
+        profile_name (str, optional): The name of the profile.
+        aws_secret_access_key (str, optional): AWS secret access key
+        aws_session_token (str, optional): AWS temporary session token
+    """
+
+    def __init__(
+        self,
+        profile_name: str = None,
+        aws_access_key_id: str = None,
+        aws_secret_access_key: str = None,
+    ):
+        if profile_name:
+            self.session = boto3.session.Session(profile_name=profile_name)
+        elif aws_access_key_id and aws_secret_access_key:
+            self.session = boto3.session.Session(
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key,
+            )
+        else:
+            self.session = boto3.session.Session()
+
+    def ls(
+        self, database: str = None, name_contains: str = None, search_text: str = None
+    ) -> List[str]:
+        """
+        Returns a list of tables in a Glue database.
+
+        Args:
+            database (str, optional): Database name.
+            name_contains (str, optional): Select by a specific string on table name.
```

Comment:
"Select by a specific string on table name." you can say sth like "match by a substring", or "substring of the table name"